src/Forms/__tests__/Checkbox.test.tsx

```diff
@@ -0,0 +1,342 @@
+import {
+  SinonSpy,
+  spy,
+} from 'sinon';
+import {
+  BoundFunction,
+  GetByText,
+  QueryByText,
+  render,
+  fireEvent,
+} from 'test-utils';
+import React, { useRef } from 'react';
+import Checkbox from 'Forms/Checkbox';
+import { strictEqual } from 'assert';
+import userEvent from '@testing-library/user-event';
+import { POSITION } from 'Forms/Label';
+import { Button } from 'Buttons';
+import { VARIANT } from 'Theme';
+
+describe('Checkbox', function () {
+  let getByText: BoundFunction<GetByText>;
+  let queryByText: BoundFunction<QueryByText>;
+  let changeSpy: SinonSpy;
+  context('when errorMessage prop is present', function () {
+    const checkboxId = 'testCheckbox';
+    const errorMessage = 'Please select this checkbox before proceeding.';
+    beforeEach(function () {
+      changeSpy = spy();
+      ({ getByText } = render(
+        <Checkbox
+          id={checkboxId}
+          checked={false}
+          label="I agree to the terms and conditions."
+          onChange={changeSpy}
+          errorMessage={errorMessage}
+          isRequired
+        />
+      ));
+    });
+    it('renders', function () {
+      const checkboxElement = document
+        .getElementById(checkboxId) as HTMLInputElement;
+      strictEqual(!!checkboxElement, true);
+    });
+    it('calls the change handler when clicked', function () {
+      fireEvent.click(document.getElementById(checkboxId));
+      strictEqual(changeSpy.callCount, 1);
+    });
+    it('renders the correct checked value', function () {
+      const checkboxElement = document
+        .getElementById(checkboxId) as HTMLInputElement;
+      strictEqual(checkboxElement.checked, false);
+    });
+    it('renders the error message', function () {
+      getByText(errorMessage, { exact: false });
+    });
+  });
+  context('when errorMessage prop is not present', function () {
+    const checkboxId = 'testCheckbox';
+    beforeEach(function () {
+      changeSpy = spy();
+      ({ queryByText } = render(
+        <Checkbox
+          id={checkboxId}
+          checked={false}
+          label="I agree to the terms and conditions."
+          onChange={changeSpy}
+          isRequired
+        />
+      ));
+    });
+    it('renders', function () {
+      const checkboxElement = document
+        .getElementById(checkboxId) as HTMLInputElement;
+      strictEqual(!!checkboxElement, true);
+    });
+    it('calls the click handler when clicked', function () {
+      fireEvent.click(document.getElementById(checkboxId));
+      strictEqual(changeSpy.callCount, 1);
+    });
+    it('renders the correct checked value', function () {
+      const checkboxElement = document
+        .getElementById(checkboxId) as HTMLInputElement;
+      strictEqual(checkboxElement.checked, false);
+    });
+    it('does not render the error message', function () {
+      strictEqual(queryByText('Please select this checkbox before proceeding.', { exact: false }), null);
```

Comment:
Would it not be better to use the `errorMessage` variable here?

Reply:
Good point, I updated this through commit 6593298.